akka-http-core/src/main/scala/akka/http/javadsl/ServerBuilder.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.ApiMayChange
+import akka.event.LoggingAdapter
+import akka.http.impl.util.JavaMapping.Implicits._
+import akka.http.javadsl.model.{ HttpRequest, HttpResponse }
+import akka.http.javadsl.settings.ServerSettings
+import akka.http.scaladsl
+import akka.http.scaladsl.{ model => sm }
+import akka.japi.function.Function
+import akka.stream.javadsl.Flow
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext
+
+trait HandlerProvider {
+  def handler(system: ClassicActorSystemProvider): Function[HttpRequest, CompletionStage[HttpResponse]]
+}
+
+/**
+ * The new API to create server bindings
+ */
+trait ServerBuilder {
+  def onPort(port: Int): ServerBuilder
+  def logTo(log: LoggingAdapter): ServerBuilder
+  def settings(settings: ServerSettings): ServerBuilder
+  def adaptSettings(f: Function[ServerSettings, ServerSettings]): ServerBuilder
+  def enableHttps(context: HttpsConnectionContext): ServerBuilder
```

Comment:
Do we also want an `enableHttps()` that enables HTTPS with the default context?

Reply:
The default context does not work as far as I know because it cannot contain any reasonable key material (and if only through ssl-config, which we'd like to get rid of).

Reply:
of course :+1: